examples/howto/optimize_rhythmic.py

```diff
@@ -0,0 +1,127 @@
+# First, we define a function that will tell the optimization routine how to
+# modify the network drive parameters. The function will take in the Network
+# object with no attached drives, and a dictionary of the parameters we wish to
+# optimize.
+
+# define set_params function and constraints
+net = jones_2009_model()
+
+
+def set_params(net, params):
+
+    # Proximal (alpha)
+    weights_ampa_p = {'L2_pyramidal': params['alpha_prox_weight'],
+                      'L5_pyramidal': 4.4e-5}
+    syn_delays_p = {'L2_pyramidal': 0.1, 'L5_pyramidal': 1.}
+
+    net.add_bursty_drive('alpha_prox',
+                         tstart=params['alpha_prox_tstart'],
+                         burst_rate=params['alpha_prox_burst_rate'],
+                         burst_std=params['alpha_prox_burst_std'],
+                         numspikes=2,
+                         spike_isi=10,
+                         n_drive_cells=10,
+                         location='proximal',
+                         weights_ampa=weights_ampa_p,
+                         synaptic_delays=syn_delays_p)
+
+    # Distal (beta)
+    weights_ampa_d = {'L2_pyramidal': params['alpha_dist_weight'],
+                      'L5_pyramidal': 4.4e-5}
+    syn_delays_d = {'L2_pyramidal': 5., 'L5_pyramidal': 5.}
+
+    net.add_bursty_drive('alpha_dist',
+                         tstart=params['alpha_dist_tstart'],
+                         burst_rate=params['alpha_dist_burst_rate'],
+                         burst_std=params['alpha_dist_burst_std'],
+                         numspikes=2,
+                         spike_isi=10,
+                         n_drive_cells=10,
+                         location='distal',
+                         weights_ampa=weights_ampa_d,
+                         synaptic_delays=syn_delays_d)
+
+###############################################################################
+# Then, we define the constraints.
+#
+# The constraints must be a dictionary of tuples where the first value in each
+# tuple is the lower bound and the second value is the upper bound for the
+# corresponding parameter.
+#
+# The following synaptic weight parameter ranges (units of micro-siemens)
+# were chosen so as to keep the model in physiologically realistic regimes.
+
+
+constraints = dict()
+constraints.update({'alpha_prox_weight': (4.4e-5, 6.4e-5),
+                    'alpha_prox_tstart': (45, 55),
+                    'alpha_prox_burst_rate': (8, 12),
+                    'alpha_prox_burst_std': (10, 25),
+                    'alpha_dist_weight': (4.4e-5, 6.4e-5),
+                    'alpha_dist_tstart': (45, 55),
+                    'alpha_dist_burst_rate': (8, 12),
+                    'alpha_dist_burst_std': (10, 25)})
+
+###############################################################################
+# Now we define and fit the optimizer.
+from hnn_core.optimization import Optimizer
+
+tstop = 300
+scale_factor = 3000
+smooth_window_len = 20
```

Comment:
Could this be messing with your ability to optimize toward a PSD peak at higher frequencies?

Reply:
I tested the code without smoothing but the smoothing doesn't seem to be the problem. It actually works really well by setting the solver to `bayesian` (the `cobyla` sovler doesn't produce good results). I'm running it now for the full 200 iterations but you can see that it's giving good results already at 15 iterations (PR description).

Reply:
I'm +1 for using `bayesian` as the default solver since we already have another function that uses cobyla

Reply:
@rythorpe we could make a note on the example to be mindful of the `scale_factor` for higher frequencies

Reply:
Both suggestions above sound good to me - thanks for looking into that @carolinafernandezp.